x/hub-genesis/keeper/genesis_transfer.go

```diff
@@ -15,32 +16,43 @@ const (
 	hubRecipient = "dym1mk7pw34ypusacm29m92zshgxee3yreums8avur"
 )
 
-// PrepareGenesisTransfer prepares the genesis transfer packet.
-// It returns the packet data if the genesis accounts are defined, otherwise it returns nil.
-// The transfer funds are escrowed explicitly in this method.
-func (k Keeper) PrepareGenesisTransfer(ctx sdk.Context, portID, channelID string) (*transfertypes.FungibleTokenPacketData, error) {
-	gAccounts := k.GetGenesisInfo(ctx).GenesisAccounts
+// PrepareGenesisBridgeData prepares the genesis bridge data.
+// Bridge data contains the genesis transfer packet data if the genesis accounts are defined, otherwise it's nil.
+// Additionally, the method returns the packet coin (if any) that will be used for the escrow.
+func (k Keeper) PrepareGenesisBridgeData(ctx sdk.Context) (types.GenesisBridgeData, sdk.Coin, error) {
```

Comment:
why u need to return `packetCoin`? it can be extracted from `data.GenesisTransfer` if needed by caller

Reply:
good point, removed it. did a separate method:
```go
// BaseCoinSupply returns the total supply of the base denom: the sum of all the genesis account amounts.
func (g GenesisInfo) BaseCoinSupply() sdk.Coin {
	amount := sdk.ZeroInt()
	for _, acc := range g.GenesisAccounts {
		amount = amount.Add(acc.Amount)
	}
	return sdk.NewCoin(g.BaseDenom(), amount)
}
```